### ISSUE TITLE
[DS Project] MIMIC dataset

### ISSUE THREAD

**Issue by anasedova** [knodle/knodle#297]
This work is a part of a master DS Project @ University of Vienna

- Analyze [MIMIC dataset](https://physionet.org/content/mimic-cxr/2.0.0/)
- Bring it to Knodle ecosystem (converting to Knodle format)
- Make experiments with different weak supervision strategies within Knodle framework

This task was created to track the progress and make sure the job is productive. Please provide a short update before our weekly meeting for each of these areas:

:trophy: What was accomplished in the previous week:
:no_entry: Problems and roadblocks:
:dart: Plans for the next week: (if you are not sure, you could add it after the meeting)

**Comment by marlenesteiner** [comment]
🏆 What was accomplished in the previous week:
The MIMIC-CXS-JPG dataset also contains images with different pixelsizes.
So I read up on the issue: All papers I found using the mimic-cxr data converted the images such that all are quadratic. So I might go with that?
Also, there is an interesting paper (https://www.medrxiv.org/content/10.1101/2021.07.30.21261225v1.full). They try different pixelsizes on the mimic-cxr data and explore how it effects the results. It is interesting to see that depending on the question sometimes the lower pixelsize, sometimes the large pixelsize achives better results.

I started working on the reports. I extracted the relevant text sections from the reports .txt files into one .csv file where every report is one row.

About my Z matrix: I thought about how to use the "synonyms for each class" (https://github.com/stanfordmlgroup/chexpert-labeler/tree/master/phrases/mention)  from chexpert-labler for my rules. There are words which are assigned to classes. Should an entry in the Z matrix should simply be 1 if a specific word appears in the report, or should it be a bit smarter, so that e.g. negations can be recognised, like: "There is no evidence of focal consolidation pleural effusion pneumothorax or frank pulmonary edema." 

⛔ Problems and roadblocks:


🎯 Plans for the next week: (if you are not sure, you could add it after the meeting)

tbd

**Comment by marlenesteiner** [comment]
🏆 What was accomplished in the previous week:
I got familiar with pytorch.
I wrote my own class to load the data such that it is compatible with the DataLoader Function. 
I finetuned a torchvision model with my data, using the weak labels. So far I only tried resnet50, but I will try different pretrained models. 
I extracted features from the fine tuned as well as from the pretrained model. 

⛔ Problems and roadblocks:


🎯 Plans for the next week: (if you are not sure, you could add it after the meeting)
tbd

**Comment by anasedova** [comment]
> 🎯 Plans for the next week: (if you are not sure, you could add it after the meeting) Write a jupiter notebook with the data preprocessing. Do some research if there are any gold labels for my dataset.
- open a PR with dataset preprocessing tutorial: Jupiter notebook + .py files

**Comment by marlenesteiner** [comment]
🏆 What was accomplished in the previous week:
This week I improved my data preprocessiong file:
*I added the computation of the T matrix: mapping rules to labels
*I did some bugfixing on the image encoding (weighted sampler, model.eval())
Then I started with the jupiter notebook. I added all the code and some descriptions, but I still need to add some more comments and exectute all snippets.
I opened a pull request. 
About the gold labels, I did not find any gold labels for mimic_cxr. However the Chexpert data contains 200 manually labeld validation data which can be accessed. 

⛔ Problems and roadblocks:
I realised that I don't have T and Z matrices yet, so I worked on that first. 
For the Z matrix I wasn't quite sure if i should add any patterns from Chexpert additionally to the mentions.
Also in which format should I save the matrices at the end of the data prep file? In one tutorial I saw sparce matrices, put the knodle input are tensors. 

🎯 Plans for the next week: 
tbd

**Comment by anasedova** [comment]
> 🎯 Plans for the next week: Complete data prep and send PR.

Do you want me to have a look at the current version of your PR, or should I rather wait till you add T and Z matrices there?

**Comment by marlenesteiner** [comment]
Maybe wait, I am planning to upload the new version by tomorrow noon. Then, there will be also outputs and more explanations why I did things how I did them.

**Comment by anasedova** [comment]
> Maybe wait, I am planning to upload the new version by tomorrow noon. Then, there will be also outputs and more explanations why I did things how I did them.

ok!

**Comment by marlenesteiner** [comment]
🏆 What was accomplished in the previous week:
Completed notebook and sent PR. 
Looked through the suggestions and applied most of them or if not, replied to the conversation.

⛔ Problems and roadblocks:
-
🎯 Plans for the next week: (if you are not sure, you could add it after the meeting)
tbd

**Comment by anasedova** [comment]
> 🏆 What was accomplished in the previous week: Completed notebook and sent PR. Looked through the suggestions and applied most of them or if not, replied to the conversation.
> 
> ⛔ Problems and roadblocks:
> 
> 🎯 Plans for the next week: Add more information in the data prep notebook. What the main idea is, why I use the weak labels...

Lemme know when the PR is ready for me to go through it once again

**Comment by marlenesteiner** [comment]
🏆 What was accomplished in the previous week:
I tried to apply my data to knodle. I think logistic regression might work for my data. Or do you have another suggestion?
⛔ Problems and roadblocks:
I had some problem with the input, since I have my X matrixes stored as np.arrays and they should be in tensorDataset format. 

🎯 Plans for the next week:

**Comment by anasedova** [comment]
Could you also upload the fine-tuned CNN to minio? thanks!

**Comment by marlenesteiner** [comment]
🏆 What was accomplished in the previous week:
Managed to speed up data downloading
Filtered records in the following way:
*only kept studies where there are two images (one lateral and one frontal)
*only kept first study of a patient
Added the test set to the data preprocessing and encode images.
⛔ Problems and roadblocks:
🎯 Plans for the next week: 
add multilabel model to knodle,
upload the CNN to minio,
concatenate embeddings of the same study
